tests/test_vcs.py

```diff
@@ -75,3 +79,21 @@ def test_dont_remove_local_clone(tmp_path):
     with Worker(src_path=src_path, dst_path=tmp_path, defaults=True) as worker:
         worker.run_copy()
     assert exists(src_path)
+
+
+def test_update_using_local_source_path_with_tilde(tmp_path):
+    src_path = vcs.clone("https://github.com/copier-org/autopretty.git")
+    fake_user_path = f"~/{'/'.join(['..'] * len(Path.home().parts))}{src_path}"
```

Comment:
Here we trick Copier by using `~` in the path but climbing up as much as needed to fall back on `/tmp`, where the temporary directory is. I believe we could have used `len(Path.home().parts) - 1`  but an extra `../` doesn't hurt.

Reply:
Comments added to the test itself.